### PR TITLE
Validatejs self exclusion

### PR DESCRIPTION
**Pull request by nazaninreihani**

### CODE REVIEW

File: src/settings/SettingsSelfExclusion.js [hunk 1]
```diff
@@ -90,109 +266,143 @@ export default class SettingsSelfExclusion extends PureComponent {
 	render() {
 		const { max_balance, max_turnover, max_losses, max_7day_turnover, max_7day_losses,
 			max_30day_turnover, max_30day_losses, max_open_bets, session_duration_limit,
-			exclude_until, timeout_until_date, timeout_until_time, success, serverError } = this.state;
-		// const wrongExcludeUntillTime = isValidTime(timeout_until_time);
+			exclude_until, timeout_until_date, timeout_until_time, success, serverError, hasError, touched, errors } = this.state;
+    this.form = document.querySelector('form#settings-self-exclusion');
```
Comment (arnabk) [inline src/settings/SettingsSelfExclusion.js]: flag 1

File: src/settings/SettingsSelfExclusion.js [hunk 2]
```diff
@@ -90,109 +266,143 @@ export default class SettingsSelfExclusion extends PureComponent {
...
+    // const wrongExcludeUntillTime = isValidTime(timeout_until_time);
 
-		return (
-			<form className="settings-self-exclusion" onSubmit={this.onFormSubmit}>
+    return (
+			<form className="settings-self-exclusion" id="settings-self-exclusion" onSubmit={this.onFormSubmit}>
```
Comment (arnabk) [inline src/settings/SettingsSelfExclusion.js]: flag 2

File: src/settings/SettingsSelfExclusion.js [hunk 3]
```diff
@@ -90,109 +266,143 @@ export default class SettingsSelfExclusion extends PureComponent {
...
+					name="max_balance"
 					label="Maximum account cash balance"
-					type="number"
+					type="text"
+					maxLength="20"
 					// hint="Once this limit is reached, you may no longer deposit."
 					defaultValue={max_balance}
 					onChange={this.onEntryChange}
 				/>
+				{touched.max_balance && errors && errors.max_balance && <ErrorMsg text={errors.max_balance[0]} />}
```
Comment (arnabk) [inline src/settings/SettingsSelfExclusion.js]: flag 3

**Review by arnabk — CHANGES_REQUESTED** [review]
@nazaninreihani I have couple of concerns with this implementation.

1. Have you considered validatejs just because I told you or you found it to be better than aprovejs?

2. This code looks like monolithic. You have to separate code an do it modular way. First create file called selfexclusion.validation.config.js and define your validation JSON in that file. This files should be imported in this file.

2. Flag 1/Flag 2 -> why do you have to use document.querySelector? This scares me when I see them in ReactJS code

3. Flag 3 -> Lot of bolierplate code here. Please do yarn add lodash/head and import head later in the code. You can replace code like `&& errors && errors.max_balance && <ErrorMsg text={errors.max_balance[0]} />` to `&& <ErrorMsg text={head((errors || {}).max_balance)} />`

4. I do not see any test code for all the changes you have done. Are you sure we do not need them?

**Comment by nazaninreihani** [comment]
@arnabk I think validate.js is more stable and suitable for us.
I fixed issues and added some tests can you please review now?

**Comment by arnabk** [comment]
@nazaninreihani All changes look good. I see that you added `enzyme-react-intl`. What for?

**Comment by nazaninreihani** [comment]
When using "Enzyme" and "React-Intl" some kind of issue happens in testing, It's for testing components that use "React-Intl's" components or injected formatting functions, so I added it to be able to use mount from "Enzyme".

**Comment by arnabk** [comment]
@nazaninreihani Is this PR complete and ready to be merged? Do you have more changes to do? Could you please move the card in trello to Review so that I could release this?

**Review by arnabk — APPROVED** [review]

**Comment by nazaninreihani** [comment]
@arnabk yes this is ready to be merged, I moved the card, Thanks :)

**Comment by nazaninreihani** [comment]
@arnabk I just found out that all fields are required only if they have value now, so please let me fix this, thanks

**Comment by nazaninreihani** [comment]
@arnabk it's now ready

**Pull request merged**